src/encoded/search.py

```
@@ -1753,7 +1753,41 @@ def set_additional_aggregations(search_as_dict, request, doc_types, extra_aggreg
     return search_as_dict
 
 
-def execute_search(search):
+def verify_search_has_permissions(request, search):
+    """
+    Inspects the search object to ensure permissions are still present on the query
+    This method depends on the query structure defined in 'set_filters'.
+
+    :param search: search object to inspect
+    :raises: HTTPBadRequest if permissions not present
+    """
+    search_dict = search.to_dict()
+    try:
+        for boolean_clause in search_dict['query']['bool']['filter']:  # should always be present
+            if 'bool' in boolean_clause and 'must' in boolean_clause['bool']:  # could be must_not,
+                possible_permission_block = boolean_clause['bool']['must']
+                for entry in possible_permission_block:
+                    if 'terms' in entry:
+                        if 'principals_allowed.view' in entry['terms']:
```

Don't you actually want to flag an error if you ever have a terms entry that doesn't have a `principals_allowed.view`? I think you need to check that any time you look at a term, no? You may want to do additional checking, as done here, but in the case that there `principals_allowed.view` is missing, I think that's not safe, but for a different reason than this code checks for. It's not safe because you didn't generate what seems to me an essential check. (I could maybe be convinced otherwise, but that's my initial position for discussion,)

---

Not necessarily - if we never find one, then yes we should throw an error - a case I have now added. I do not see a code path where you could actually remove that part of the query altogether, but better to be safe.